### PR TITLE
[libvpx] Fix libvpx compilation (install process)

### PR DESCRIPTION
**Pull request by talregev**
Fix libvpx compilation (install process)
compile on windows (x86, x64)
Fix Cannot find pdb (line 104 in profile).

### CODE REVIEW

**Comment by JackBoosY** [comment]
```
-- Installing: D:/packages/libvpx_x64-windows/lib/vpxmd.lib
CMake Error at ports/libvpx/portfile.cmake:104 (file):
  file INSTALL cannot find
  "D:/buildtrees/libvpx/x64-windows/x64/Release/vpxmd.pdb": File exists.
Call Stack (most recent call first):
  scripts/ports.cmake:142 (include)
```
Can you double confirm your changes?

**Comment by talregev** [comment]
> ```
> -- Installing: D:/packages/libvpx_x64-windows/lib/vpxmd.lib
> CMake Error at ports/libvpx/portfile.cmake:104 (file):
>   file INSTALL cannot find
>   "D:/buildtrees/libvpx/x64-windows/x64/Release/vpxmd.pdb": File exists.
> Call Stack (most recent call first):
>   scripts/ports.cmake:142 (include)
> ```
> 
> Can you double confirm your changes?

Is it from your pc?
On one of the computer libvpx compile without the change. I compile only the libvpx
Other computer I need to do the change. I compile libvpx as part of ffmpeg[all-gpl]
Maybe if I compile libvpx part of other libs, it happen?
Can you investigate it more? 

Both computers running windows 10.

**Comment by JackBoosY** [comment]
@talregev  From our pipeline test, Windows10 with Visual Studio 2019.

**Comment by JackBoosY** [comment]
@talregev  What triplet do you use?

**Comment by talregev** [comment]
both x86-windows x64-windows

**Comment by mcmtroffaes** [comment]
I haven't seen this happen on my CI (with x64-windows, x64-windows-static, and x64-windows-static-md); installation and compilation works fine. For the record, logs are here (for an ffmpeg master build, but otherwise the portfiles are the same, and in particular its the same for libvpx):

https://ci.appveyor.com/project/mcmtroffaes/ffmpeg-msvc-build/build/job/3yn0l4c3ge9fwkiy (x64-windows)
https://ci.appveyor.com/project/mcmtroffaes/ffmpeg-msvc-build/build/job/stavyb3m9i23my7w (x64-windows-static)
https://ci.appveyor.com/project/mcmtroffaes/ffmpeg-msvc-build/build/job/yr0nuflv8k8e4j0a (x64-windows-static-md)

This seems vaguely related to #12706 where the names were fixed to ensure the pdb files were found when compiling other apps against ffmpeg: the pdb files cannot be renamed after install.

**Comment by talregev** [comment]
> This seems vaguely related to #12706 where the names were fixed to ensure the pdb files were found when compiling other apps against ffmpeg: the pdb files cannot be renamed after install.

I think it a good lead and I would start from there. Did you try / ci compile `vcpkg ffmpeg[all-gpl]`?
I think when it compile ffmep with all the option, it also compile an apps that using ffmpeg, and maybe that why it rename the pdb. if it that the case, my suggestion is that the pdb always need to be rename

**Comment by JackBoosY** [comment]
I think CRT linkage affected it.
Can you please provide `vpx.vcxproj` in _VCPKG_ROOT/buildtrees/libvpx/x86-windows_ ?

**Comment by mcmtroffaes** [comment]
> Did you try / ci compile `vcpkg ffmpeg[all-gpl]`?

That's a good suggestion. I just tested `vcpkg ffmpeg[core,ffmpeg,libvpx]` on x64-windows and I can confirm that this works as expected at least on my local setup, with the ffmpeg binary able to encode with the libvpx-vp9 codec. Note that building ffmpeg (with whichever feature set) will not change the libvpx build on vcpkg: the ffmpeg build process does not influence the build process of its dependencies (such as libvpx).

**Comment by talregev** [comment]
I preproduce the problem with small example. 
`vcpkg install ffmpeg[core,ffmpeg,vpx]:x64-windows`
[vpx.vcxproj and logs.zip](https://github.com/microsoft/vcpkg/files/6326741/vpx.vcxproj.and.logs.zip)

**Comment by talregev** [comment]
Maybe the problem is from the things I install on visual studio. like mfc and clang support?

**Comment by talregev** [comment]
@JackBoosY any news? 
Do you need more information?

**Comment by JackBoosY** [comment]
@talregev  Nope.

**Comment by JackBoosY** [comment]
@talregev  Can you test whether my changes is correct?

**Review by JackBoosY — APPROVED** [review]

File: ports/ffmpeg/CONTROL [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Source: ffmpeg
 Version: 4.3.2
-Port-Version: 3
+Port-Version: 4
```
Comment (JackBoosY) [inline ports/ffmpeg/CONTROL:3]:
```suggestion
Port-Version: 5
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by mcmtroffaes** [comment]
Thanks @JackBoosY! For the record, I've just tested this pr by compiling an application against ``ffmpeg[core,avcodec,vpx]`` on the x64-windows and x64-windows-static triplets, and I can confirm that it works like charm, besides also cleaning things up for libvpx detection on ffmpeg. Fingers crossed it also fixes @talregev 's issue.

File: versions/f-/ffmpeg.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "f4fe9fdcd47745dee73dfecc23d4abd0a5515cb0",
```
Comment (JackBoosY) [inline versions/f-/ffmpeg.json:4]:
```suggestion
      "git-tree": "68f6d071ba23bf8ad28a12a507c8089a47a76d73",
```

**Review by JackBoosY — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by strega-nil** [comment]
LGTM! Thanks @talregev

**Comment by BillyONeal** [comment]
Unfortunately it doesn't look like this fixed it, in https://github.com/microsoft/vcpkg/pull/17331 with this change applied we still see:

```
CMake Error at ports/libvpx/portfile.cmake:104 (file):
  file INSTALL cannot find
  "D:/buildtrees/libvpx/arm-uwp/ARM/Release/vpx/vpx.pdb": File exists.
```

**Comment by BillyONeal** [comment]
I think I have a fix, will put into the VM update PR

**Comment by Hoikas** [comment]
A user of my project is reporting this same failure even after this PR.

**Comment by mcmtroffaes** [comment]
I've started hitting this now as well on appveyor CI (not locally); https://github.com/microsoft/vcpkg/commit/91542464a564bfad42fcf2fa2fbafa3cab929a54 also fixed it for me. Perhaps something to do with the version of Visual Studio.

**Comment by Hoikas** [comment]
This issue is now being tripped on GitHub Actions CI. Is there any chance the fix can be fast-tracked @BillyONeal?

**Comment by BillyONeal** [comment]
> This issue is now being tripped on GitHub Actions CI. Is there any chance the fix can be fast-tracked @BillyONeal?

The VM update just landed, try again?

**Comment by Hoikas** [comment]
That seems to have fixed it, thanks!

**Comment by JackBoosY** [comment]
@Hoikas @mcmtroffaes  Fixed.

**Comment by mcmtroffaes** [comment]
Yup, happy to confirm the fix in master now. Thanks a bunch!

**Pull request merged**